### PR TITLE
Add Project + lead Query

### PR DESCRIPTION
**Pull request by thenav56**
Addresses
- #848
- #847
- #845 
## Changes

- Add Project + lead Query
- Setup basic permissions for Graphql.
- Using poetry. (Remove pipenv)
- Add lead mutations.

Mention related users here if any.

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] n+1 queries
- [x] flake8 issues
- [x] `print`
- [x] typos
- [x] unwanted comments

## This PR contains valid:

- [x] tests
- [x] permission checks (tests here too)

### CODE REVIEW

File: apps/lead/schema.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    class Meta:
+        model = Lead
+        fields = (
+            'id', 'title', 'created_by', 'created_at', 'modified_by', 'modified_at',
+            'project', 'lead_group', 'assignee', 'published_on',
+            'source_type', 'priority', 'confidentiality', 'status',
+            'text', 'url', 'website', 'attachment', 'emm_entities'
+        )
+
+    project = graphene.ID(source='project_id')
```
Comment (k9845) [inline apps/lead/schema.py:75]: why project field here as well, since we have in fields?

File: apps/project/schema.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    # NOTE: This is a custom feature
+    # see: https://github.com/eamigo86/graphene-django-extras/compare/graphene-v2...the-deep:graphene-v2
+    @staticmethod
+    def get_custom_node(queryset, info, id):
+        try:
+            project = Project.get_for_gq(info.context.user).get(pk=id)
+            info.context.set_active_project(project)
+            return project
+        except Project.DoesNotExist:
+            return None
```
Comment (k9845) [inline apps/project/schema.py]: What does this method do?

Comment (thenav56) [reply apps/project/schema.py]: There is a comment there.

    # NOTE: This is a custom feature
    # see: https://github.com/eamigo86/graphene-django-extras/compare/graphene-v2...the-deep:graphene-v2

File: apps/lead/schema.py (hunk 1 above)
Comment (thenav56) [reply apps/lead/schema.py:75]: This is to force the field to be ID instead of ProjectType.

**Review by k9845 — APPROVED** [review]

**Pull request merged**